blockchain/v2/reactor.go

```diff
@@ -389,6 +389,10 @@ func (r *BlockchainReactor) demux(events <-chan Event) {
 			case scSchedulerFail:
 				r.logger.Error("Scheduler failure", "err", event.reason.Error())
 			case scPeersPruned:
+				// Remove peers from the processor.
+				for _, peerID := range event.peers {
+					r.processor.send(scPeerError{peerID: peerID, reason: errors.New("peer was pruned")})
```

Comment:
I think it's kind of weird to be passing errors around as a way to remove peers. Could we add a new message (unless there's one already) that simply triggers a peer to be removed without looking like an error?

Reply:
I think `scPeerError` is fine in this context. scheduler signals that the peer is bad (slow, disconnected, malicious). reactor resends this event to processor, which removes the peer from its state.